packages/analytics-browser-test/test/index.test.ts

```diff
@@ -1232,8 +1238,15 @@ describe('integration', () => {
             }, {}),
           )
             .map((c) => c.sort((a, b) => a.time - b.time))
-            .map((group) => group.map((c) => ({ eventType: c.eventType, userId: c.userId })));
-
+            .map((group) => group.map((c) => ({ eventType: c.eventType, userId: c.userId, deviceId: c.deviceId })));
+          const deviceIds = compactEventsBySessionAndTime.flat().map((c) => c.deviceId);
+          expect(deviceIds[0]).toEqual(deviceIds[1]);
+          expect(deviceIds[1]).toEqual(deviceIds[2]);
+          expect(deviceIds[2]).toEqual(deviceIds[3]);
+          expect(deviceIds[3]).toEqual(deviceIds[4]);
+          expect(deviceIds[4]).toEqual(deviceIds[5]);
+          expect(deviceIds[5]).toEqual(deviceIds[6]);
+          expect(deviceIds[6]).not.toEqual(deviceIds[7]);
```

Comment:
Asserts that only the last event (Which is a session start for a new user) has a different device id